sdk/app_cpu1/common/drv/timing_manager.c

```diff
@@ -208,16 +213,75 @@ double timing_manager_get_tick_delta(void)
 }
 
 /*
- * Set the user-defined ratio for the event qualifier
+ * Gets the expected time (in us) between ISR calls based on the user's switching frequency,
+ * PWM event sub-ratio, and enabled sensors
+ */
+double timing_manager_expected_tick_delta(void)
+{
+    double fsw = pwm_get_switching_freq();
+
+    // The no-sensor-time is the expected time in us between scheduler interrupts
+    // with no sensors enabled. This is just the switching period multiplied by the sub-ratio
+    double no_sensor_time = (now_ratio / fsw) * 1e6;
```

Comment:
This assumes the PWM carrier event is only once per period (the default), but a user could configure "double update rate" i.e., trigger events at both high and low.